### ISSUE TITLE
Generating unit tests for current function generates tests for other functions, too

### ISSUE THREAD

**Issue by mrwonko** [Microsoft/vscode-go#1483]
If you have two functions where one's name is a prefix of the other's and you generate unit tests for the shorter one, tests are also generated for the other one:
```Go
func foo()    {}
func fooBar() {}
```
Upon executing "Go: Generate Unit Tests for Function" for `foo()` I get a message saying "Generated: Test_foo, Test_fooBar" and the following tests:
```Go
func Test_foo(t *testing.T) {
	tests := []struct {
		name string
	}{
	// TODO: Add test cases.
	}
	for range tests {
		t.Run(tt.name, func(t *testing.T) {
			foo()
		})
	}
}

func Test_fooBar(t *testing.T) {
	tests := []struct {
		name string
	}{
	// TODO: Add test cases.
	}
	for range tests {
		t.Run(tt.name, func(t *testing.T) {
			fooBar()
		})
	}
}
```

**Comment by ramya-rao-a** [comment]
Thanks for reporting this issue @mrwonko!

Looks like the `gotests` tool was expected a regular expression for the function names. Since we passed just the function names, anything that matched got picked up.

Fixed it with https://github.com/Microsoft/vscode-go/commit/193516764c80a9289d278e76898a924ffc9d6a52

The fix will be out in the next update to the Go extension which should be out next week
